lib/assigner.rb

```diff
@@ -514,26 +514,35 @@ def moderator_post_unassign_action_code(assignment)
 
   def topic_same_assignee_and_details(assign_to, type, note, status)
```

Comment:
Could we merge `topic_same_assignee_and_details` and `post_same_assignee_and_details` into just one method? Seems like there's a lot of repetition for no good reason.